torch/backends/cudnn/rnn.py

```diff
@@ -190,7 +196,11 @@ def forward(fn, input, hx, weight, output, hy):
         handle = cudnn.get_handle()
         fn.datatype = cudnn._typemap[input.type()]
         is_input_packed = fn.batch_sizes is not None
-
+        if is_input_packed:
+            warnings.warn(
+                "Warning: persistent algorithm not supported for variable length input."
+                "Switching to standard")
+            fn.persistent = False  # persistent algo is not supported for variable length input
```

Comment:
Wow, this is easily twice as fast for my current use case (the whole model -- the LSTMs are probably 5x faster). Thanks Natalia! Now we just need more P100s... :)
(Also, I'm okay with setting the flag every time but defaulting to use persistent mode for cases where it's obviously better is probably worth it)